_posts/2017-10-17-faf-in-openshift.markdown

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title:  "FAF in OpenShift"
+date:   2017-10-17 12:00:00 +0100
+author: mkutlak
+visible: 1
+categories: FAF container OpenShift
+---
+
+Guide on how to deploy your own [FAF](https://www.github.com/abrt/faf) in [OpenShift](https://www.openshift.com/).
+
+# Deploy FAF in OpenShift
+
+The following text will guide you through the process of deploying FAF in OpenShift and using it to collect crash reports from your machines.
+There are two options how you can do it. The first option is to use the command line, and the second one is to use the web interface.
+
+## Prerequisites
+ - [OpenShift Origin](https://github.com/openshift/origin/releases)
+ - Cluster admin rights for editing security context constraints, and managing
+   volumes.
+
+
+## Command Line
+
+1. Create a new project.
+```
+oc new-project <project_name>
+```
+
+2. Add the FAF application. We've created docker image with FAF, that you can find on
+[Docker Hub](https://hub.docker.com/r/abrt/faf-image).
+```
+oc new-app abrt/faf-image --name <app_name>
+```
+This command will pull the image from docker hub and then deploy it.
+
+3. Edit Security Context Constraints. <br><br>
+The deployment won't succeed because the FAF container requires usage of the root user, but default
+configuration doesn't allow it. To allow the container to use of root, create a new service account
+and add it to **anyuid's** security context constraints.
+```
+oc create serviceaccount <sa_username>
+oc adm add-scc-to-user anyuid -z <sa_username>
+oc patch dc/<app_name> --patch '{"spec":{"template":{"spec":{"serviceAccountName": "<sa_username>"}}}}'
+```
+Changing deployment configuration will trigger new container deployment. This time, it should succeed,
+and the FAF should be running and accessible through ```http://<pod_ip>/faf```.
+<br><br>
+To get IP address of the pod:
+```
+{% raw %}oc get pod <pod_name> --template='PodIP: {{.status.podIP}}{{"\n"}}'{% endraw %}
```

Comment:
{% raw %} {% endraw %} are likely left overs

Reply:
It stops Jekyll from processing {{ .... }} as tags.

Reply:
This is only internal IP. It will be only available from master on nodes.
You should run

oc expose dc <app_name>

This will create service and generate hostname which will be reachable from outside network.

Reply:
Err. This will not print the hostname. After the expose you have to run one more command:

oc get route <app_name>

which will print the hostname available from outside of network.